kombu/transport/SQS.py

```diff
@@ -191,15 +194,27 @@ def _delete(self, queue, *args, **kwargs):
 
     def _put(self, queue, message, **kwargs):
         """Put message onto queue."""
-        q = self._new_queue(queue)
-        m = Message()
-        m.set_body(dumps(message))
-        q.write(m)
+        q_url = self._new_queue(queue)
+        kwargs = {'QueueUrl': q_url,
+                  'MessageBody': AsyncMessage().encode(dumps(message))}
+        if queue.endswith('.fifo'):
+            if 'MessageGroupId' in message['properties']:
+                kwargs['MessageGroupId'] = \
+                    message['properties']['MessageGroupId']
+            else:
+                kwargs['MessageGroupId'] = 'default'
+            if 'MessageDeduplicationId' in message['properties']:
+                kwargs['MessageDeduplicationId'] = \
+                    message['properties']['MessageDeduplicationId']
+            else:
+                kwargs['MessageDeduplicationId'] = str(uuid.uuid4())
+        self.sqs.send_message(**kwargs)
 
     def _message_to_python(self, message, queue_name, queue):
-        payload = loads(bytes_to_str(message.get_body()))
+        payload = loads(bytes_to_str(message['Body']))
```

Comment:
@alukach you can see that the base64 encoding is related to the Message implementation of boto [here](https://github.com/boto/boto/blob/master/boto/sqs/message.py#L154).